woocommerce/woocommerce
# Fix selectors for quantity element in variable products

### All Submissions:


### Changes proposed in this Pull Request:

Closes #36855 to fix bug introduced by PR #36087.  
Also addresses original issue #36086.

***Edit 2023-02-20 (added details from [commit message](https://github.com/woocommerce/woocommerce/pull/36871/commits/e0aa4827b8975ab80c1af7be5aba7cb7819ddfd9)):***
WooCommerce 7.4.0 changed the jQuery selectors for the quantity input in variable products.

This PR reinstates the previous logic so that `$qty` points to the div wrapper, and `$qty_input` points to the input element inside.

Otherwise, as is in the affected version, `$qty_input` is undefined and the min/max attributes are not updated.

It also addresses the original issue #36086 by restricting the selector of the div wrapper to only be the one containing `input[name=quantity]` (the `.qty` class filter is also kept to avoid any potential malfunctions of existing code).

### How to test the changes in this Pull Request:


#### Quantity selector works properly (adapted from steps to reproduce in #36855):
1. Create a variable product with some variations.
1. Set a variation to "manage stock".
1. Set the variation's "Stock quantity".
1. Set the variation to "Do not allow" backorders.
1. View the product from the front end.
1. Pick the variation.
1. Try increasing the quantity higher than what you set the stock quantity to.
1. The input should now enforce the max limit so the value should not change above the set quantity.

#### Quantity selector does not affect other inputs (adapted from testing instructions in #36087):

Add the following snippet, which adds a new quantity input to the single variation:

```php
function kia_test_single_variation_qty() {
	echo '<div class="quantity"><input class="qty" type="number" name="tacos" /></div>';
}
add_action( 'woocommerce_before_single_variation', 'kia_test_single_variation_qty' );
```

1. Go to the variable product above.
1. Select a variation by selecting the attributes necessary.
1. Increase the rogue quantity input's value.
1. Change to another variation.
1. The main quantity input's value should stay the same.

### Other information:


### FOR PR REVIEWER ONLY:


cc @barryhughes
